src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -574,8 +574,12 @@ def send_raw_request(cli_ctx, method, uri, headers=None, uri_parameters=None,  #
     if not skip_authorization_header and uri.lower().startswith('https://'):
         if not resource:
             endpoints = cli_ctx.cloud.endpoints
+            from azure.cli.core.cloud import CloudEndpointNotSetException
             for p in [x for x in dir(endpoints) if not x.startswith('_')]:
-                value = getattr(endpoints, p)
+                try:
+                    value = getattr(endpoints, p)
+                except CloudEndpointNotSetException:
+                    continue
```

Comment:
When users want to use the endpoint which is not set, can your code still throw correct message?

Reply:
Yes. At L593.